src/irmin-pack/pack.ml

```diff
@@ -288,20 +304,24 @@ module File (K : Irmin.Hash.S) = struct
       | true -> ()
       | false ->
           Log.debug (fun l -> l "[pack] append %a" pp_hash k);
+          IO.sync t.pack.block;
```

Comment:
This is readlly not satisfactory, for at the moment that's the only way I know of to hit the elements that were just added, but of which we can't access the hash on disk yet.

Reply:
Solved by adding an offset cache for the staging area.